engine/pom.xml

```
@@ -36,4 +36,32 @@
             <scope>test</scope>
         </dependency>
     </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.moditect</groupId>
+                <artifactId>moditect-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>add-module-infos</id>
+                        <configuration>
+                            <module>
+                                <moduleInfo>
+                                    <name>${java.module.name}</name>
+                                    <exports>
+                                        org.hibernate.search.*.impl to org.hibernate.search.util.common, org.hibernate.search.engine, org.hibernate.search.backend.elasticsearch, org.hibernate.search.backend.elasticsearch.aws, org.hibernate.search.mapper.pojo, org.hibernate.search.mapper.javabean, org.hibernate.search.mapper.orm, org.hibernate.search.batch.jsr352.core, org.jboss.logging;
```

You definitely shouldn't need to export `impl` packages here. It's only necessary in the `hibernate-search-util-common` module.

```suggestion
                                        !org.hibernate.search.*.impl;
```